experimental/traffic-portal/src/app/api/server.service.ts

```diff
@@ -186,4 +187,33 @@ export class ServerService extends APIService {
 
 		return this.put(`servers/${id}/status`, {offlineReason, status}).toPromise();
 	}
+
+	/**
+	 * Creating new Status.
+	 *
+	 * @param status The status to create.
+	 * @returns The created status.
+	 */
+	public async createStatus(status: RequestStatus): Promise<ResponseStatus> {
+		return this.post<ResponseStatus>("statuses", status).toPromise();
+	}
+
+	/**
+	 * Updates status Details.
+	 *
+	 * @param status The status to update.
+	 * @returns The updated status.
+	 */
+	public async updateStatusDetail(status: ResponseStatus): Promise<ResponseStatus> {
+		return this.put<ResponseStatus>(`statuses/${status.id}`, status).toPromise();
+	}
+
+	/**
+	 * Deletes an existing Status.
+	 *
+	 * @param id The Status ID
+	 */
+	public async deleteStatus(id: number | ResponseStatus): Promise<ResponseStatus> {
+		return this.delete<ResponseStatus>(`statuses/${id}`).toPromise();
+	}
```

Comment:
This method won't work for all call signatures; if you pass in a `ResponseStatus` it will attempt to use the API endpoint `/statuses/%5Bobject%20Object%5D` which does not exist because `%5Bobject%20Object%5D` is not a valid Status ID.